apis/system-settings.md

```diff
@@ -0,0 +1,58 @@
+---
+description: Deep linking to System Settings
+---
+
+# System Settings
+
+Some of the features of your application may depend on system level settings or services, such as Internet connectivity or app level access to the network. While your app cannot directly change those settings, it can prompt users to change them by linking to specific panes in the System Settings.
+
+elementary OS implements a special `settings://` URI scheme for linking to System Settings. The links lead to specific settings panes. You can find most commonly used ones in the `Granite.SettingsUri` namespace in the Granite library.
+
+## Opening a link
+
+First, make sure you've included Granite in the build dependencies declared in your `meson.build` file:
+
+{% code title="meson.build" %}
+```coffeescript
+executable(
+    meson.project_name(),
+    'src' / 'Application.vala',
+    dependencies: [
+        dependency('granite-7'),
+        dependency('gtk4')
+    ],
+    install: true
+)
+```
+{% endcode %}
+
+Then you can open Network settings by launching `Granite.SettingsUri.NETWORK`:
+
+```vala
+var button = new Gtk.Button.with_label ("Change network settings");
+button.clicked.connect (() => {
+    try {
+        AppInfo.launch_default_for_uri (Granite.SettingsUri.NETWORK, null);
+    } catch (Error e) {
+        warning ("Failed to open network settings: %s", e.message);
+    }
```

Comment:
We actually wanna use [Gtk.UriLauncher](https://valadoc.org/gtk4/Gtk.UriLauncher.html) here because `AppInfo`doesn't work with focus stealing prevention, so the settings window won't get focused correctly etc

Reply:
I didn't know about the `Gtk.UriLauncher`. I've updated the code snippet. I hope it's ok, I still have Gtk 4.6.6 on this machine so I wasn't able to run the new code.

Reply:
I would recommend building a little Flatpak app in that case! I have a super minimal sample you can build off at https://github.com/vala-lang/examples

Reply:
Hm... Looks like elementary runtime version 7.2 has too old vala for this to work.

Current vala version in 7.2 is `0.56.4`.
Vapi for Gtk.UriLauncher was introduced in [this commit](https://gitlab.gnome.org/GNOME/vala/-/commit/3b0d9a70008199fd59f89aa0ad32eca2b55d2613), which was released in vala version `0.56.5`.
I tested the uri launching snippet with gnome runtime 44 (which has vala `0.56.7`) and it worked as expected.

I guess all of this will ultimately be ok, since we need to wait with publishing this docs section until elementary runtime 7.3 is out. We can include both the new granite and the new vala (with appropriate gtk 4.10 vapi) in that upgrade. And then the code snippet will work out of the box. I've expanded the paragraph about dependencies and added a flatpak manifest snippet with runtime version 7.3.